From 4a65b1b278130a51341c84bce5c843faa2ae2805 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:04:54 -0400
Subject: [ethernet] qlcnic: Use firmware recommended dump capture mask as default

Message-id: <1408626422-3941-11-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88546
O-Subject: [PATCH RHEL 7.1 010/138] qlcnic: Use firmware recommended dump capture mask as default
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shahed Shaikh <shahed.shaikh@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 7010bb65ce01278abe6709fe90407183abc6cbef

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
index 811b6c9..4b3763f 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
@@ -1404,7 +1404,6 @@ struct qlcnic_esw_statistics {
  struct __qlcnic_esw_statistics tx;
 };
 
-#define QLCNIC_DUMP_MASK_DEF  0x1f
 #define QLCNIC_FORCE_FW_DUMP_KEY 0xdeadfeed
 #define QLCNIC_ENABLE_FW_DUMP  0xaddfeed
 #define QLCNIC_DISABLE_FW_DUMP  0xbadfeed
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_minidump.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_minidump.c
index 79e54ef..b7871fe 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_minidump.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_minidump.c
@@ -1082,7 +1082,10 @@ flash_temp:
  }
 
  tmpl_hdr = ahw->fw_dump.tmpl_hdr;
- tmpl_hdr->drv_cap_mask = QLCNIC_DUMP_MASK_DEF;
+ tmpl_hdr->drv_cap_mask = tmpl_hdr->cap_mask;
+ dev_info(&adapter->pdev->dev,
+   "Default minidump capture mask 0x%x\n",
+   tmpl_hdr->cap_mask);
 
  if ((tmpl_hdr->version & 0xfffff) >= 0x20001)
   ahw->fw_dump.use_pex_dma = true;
-- 
1.7.1